### ISSUE TITLE
first boot after coreos-install fails on veritysetup

### ISSUE THREAD

**Issue by redbaron** [coreos/bugs#2529]
# Issue Report #

### Container Linux Version ###

1911.3.0

### Environment ###

Baremetal

### Actual Behavior ###

veritysetup fails during boot, inspecting emergency shell shows that `e2size /dev/<device>` returns "Success" string, not numerical value.  It seems that e2size in initrd is NOT https://github.com/coreos/seismograph/blob/master/src/e2size/e2size.c

### Reproduction Steps ###

  1. ...
  2. ...

### Other Information ###

Same deployment procedure works fine in VMWare environment

**Comment by redbaron** [comment]
we provision using iPXE,  where network boot runs coreos-install and then reboots. Second boot was exactly what was failing.

I rebooted over iPXE again and it now passes verity-setup. How strange!  When it was failing, I was using emergency shell to get some intel, it it was an error from verity-setup's

`/sbin/veritysetup create usr --hash-offset="$(e2size /dev/disk/by-partuuid/...) ..."`

when I ran `e2size /dev/didk/by-partuuid/....` I got result `Success`

**Comment by lucab** [comment]
Yes, that was also my suspicion. An exit-code of 1 would confirm that. I'm still unsure about the root-cause of this one-off failure though. Was the same disk used for a previous Linux installation without any zero-ing in-between?

**Comment by redbaron** [comment]
> An exit-code of 1 would confirm that.

because it is in a shell substitution, that exit code was ignored and it tried to run verity-setup anyway

> Was the same disk used for a previous Linux installation without any zero-ing in-between?

doesn't coreos-install script erase whole disk anyway?
